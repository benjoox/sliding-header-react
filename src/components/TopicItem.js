import React, { useRef, useEffect } from 'react'
import { func, number, string } from 'prop-types'

export default function TopicItem({
  topic,
  index,
  setElementWidth,
  className,
}) {
  const liRef = useRef()
  useEffect(() => {
    const { current } = liRef
    if (current) {
      setElementWidth(index, current.getBoundingClientRect().width)
    }
  }, [liRef, index])
  const classname = className ? `topic-item ${className}` : 'topic-item'
  return (
    <li ref={liRef} className={classname}>
      <a href="#">{topic}</a>
    </li>
  )
}

TopicItem.defaultProps = {
  className: null,
}

TopicItem.protoType = {
  topic: string.isRequired,
  index: number.isRequired,
  setElementWidth: func.isRequired,
  className: string,
}
