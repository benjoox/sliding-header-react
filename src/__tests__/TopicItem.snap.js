import React from 'react'
import renderer from 'react-test-renderer'
import TopicItem from '../components/TopicItem'

it('TopicItem  renders correctly', () => {
  const props = {
    topic: 'Debate X',
    index: 1,
    setElementWidth: () => {},
    className: 'collapsed',
  }
  const tree = renderer.create(<TopicItem {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
