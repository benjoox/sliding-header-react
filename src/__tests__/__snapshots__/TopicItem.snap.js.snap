// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicItem  renders correctly 1`] = `
<li
  className="topic-item collapsed"
>
  <a
    href="#"
  >
    Debate X
  </a>
</li>
`;
